clients/src/test/java/org/apache/kafka/common/metrics/JmxReporterTest.java

```diff
@@ -48,7 +48,8 @@ public void testJmxRegistration() throws Exception {
             assertTrue(server.isRegistered(new ObjectName(":type=grp2")));
             assertEquals(0.0, server.getAttribute(new ObjectName(":type=grp2"), "pack.bean2.total"));
 
-            metrics.removeMetric(metrics.metricName("pack.bean1.avg", "grp1"));
+            KafkaMetric metric = metrics.removeMetric(metrics.metricName("pack.bean1.avg", "grp1"));
+            assertEquals(0, metrics.getCount(metric));
```

Comment:
It seems like we can test this more easily by adding a new test where we instantiate `JxmReporter` and call its methods directly.

Reply:
The modified test covers life cycle of JxmReporter.
IMHO this should be enough.

Reply:
@ijuma 
Waiting for your comment before making further change to the test.

Reply:
The point is avoiding `Metrics.getCount` which is suspect code (it assumes all metric reporters are `JmxReporter` instances, for example). It would be much better if we don't have to add any methods to `Metrics` to test `JmxReporter`.